tests/mock_vthttpserver.py

```diff
@@ -0,0 +1,258 @@
+        resource = \
+            urlparse.parse_qs(parsed_url.query)[VirusTotalApiRequestCallback.PARAM_RESOURCE][0]
+        if resource == SAMPLE_FILE:
+            return MessageUtils.dict_to_json(SAMPLE_FILE_REPORT, pretty_print=False)
+        return self.bad_param(VirusTotalApiRequestCallback.PARAM_RESOURCE, resource)
+
+
+    def file_rescan_cmd(self, parsed_url):
+        resource = \
+            urlparse.parse_qs(parsed_url.query)[VirusTotalApiRequestCallback.PARAM_RESOURCE][0]
+        if resource == SAMPLE_FILE:
+            return MessageUtils.dict_to_json(SAMPLE_FILE_RESCAN, pretty_print=False)
+        return self.bad_param(VirusTotalApiRequestCallback.PARAM_RESOURCE, resource)
+
+
+    def ip_report_cmd(self, parsed_url):
+        ip_address = \
+            urlparse.parse_qs(parsed_url.query)[VirusTotalApiRequestCallback.PARAM_IP][0]
+        if ip_address == SAMPLE_IP:
+            return MessageUtils.dict_to_json(SAMPLE_IP_ADDRESS_REPORT, pretty_print=False)
+        return self.bad_param(VirusTotalApiRequestCallback.PARAM_IP, ip_address)
+
+
+    def url_report_cmd(self, parsed_url):
+        url = \
+            urlparse.parse_qs(parsed_url.query)[VirusTotalApiRequestCallback.PARAM_RESOURCE][0]
+        if url == SAMPLE_URL:
+            return MessageUtils.dict_to_json(SAMPLE_URL_REPORT, pretty_print=False)
+        return self.bad_param(VirusTotalApiRequestCallback.PARAM_RESOURCE, url)
+
+
+    def url_scan_cmd(self, parsed_url):
+        url = \
+            urlparse.parse_qs(parsed_url.query)[VirusTotalApiRequestCallback.PARAM_URL][0]
+        if url == SAMPLE_URL:
+            return MessageUtils.dict_to_json(SAMPLE_URL_SCAN, pretty_print=False)
+        return self.bad_param(VirusTotalApiRequestCallback.PARAM_URL, url)
+
+
+    # Needs to return ERROR 204 HEADER!
+    #@staticmethod
+    #def rate_limit_exceeded(cmd_string):
+    #    # Needs to return ERROR 204 HEADER!
+    #    return None
+
+
+    @staticmethod
+    def bad_param(param_name, param_val):
+        return MessageUtils.dict_to_json(
+            {
+                "unit_test_bad_param_name": param_name,
+                "unit_test_bad_param_val": param_val
+            },
+            pretty_print=False
+        )
+
+    @staticmethod
+    def unknown_call(path):
+        return MessageUtils.dict_to_json(
+            {
+                "unit_test_error_unknown_api": path
+            },
+            pretty_print=False
+        )
+
+
+class MockServerRunner(object):
+
+    def __init__(self):
+        self.server_name = "mockvtserver"
+        self.mock_server_port = 0
+        self.mock_server = None
+        self.mock_server_address = ""
+
+    def __enter__(self):
+        self.mock_server_address, self.mock_server_port = get_free_port()
+        self.mock_server = TCPServer(
+            ('localhost', self.mock_server_port),
+            MockVtServerRequestHandler
+        )
+        #self.mock_server.socket = ssl.wrap_socket(
```

Comment:
The problem with using https lies with the Python requests library we're using. It was problematic to use with the ssl features of SimpleHTTPServer that I use in unit tests in other projects - and I figured that because none of the SSL aspects in use here are actually dealt with in the dxlvtapiservice code, we could just assume that part works.

(In the dxleposervice project, I cared more about the SSL because it was a mandatory part of the request library calls being used, where here it is not)